Emby.Server.Implementations/Library/PathExtensions.cs

```diff
@@ -47,5 +48,84 @@ public static class PathExtensions
+        /// <exception cref="ArgumentNullException"><paramref name="path" />, <paramref name="newSubPath" /> or <paramref name="newSubPath" /> is empty.</exception>
+        public static bool TryReplaceSubPath(this string path, string subPath, string newSubPath, [NotNullWhen(true)] out string? newPath)
+        {
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+
+            if (string.IsNullOrWhiteSpace(subPath))
+            {
+                throw new ArgumentNullException(nameof(subPath));
+            }
+
+            if (string.IsNullOrWhiteSpace(newSubPath))
+            {
+                throw new ArgumentNullException(nameof(newSubPath));
+            }
+
+            char oldDirectorySeparatorChar;
+            char newDirectorySeparatorChar;
+            // True normalization is still not possible https://github.com/dotnet/runtime/issues/2162
+            // The reasoning behind this is that a forward slash likely means it's a Linux path and
+            // so the whole path should be normalized to use / and vice versa for Windows (although Windows doesn't care much).
+            if (newSubPath.Contains('/', StringComparison.Ordinal))
+            {
+                oldDirectorySeparatorChar = '\\';
+                newDirectorySeparatorChar = '/';
+            }
+            else
+            {
+                oldDirectorySeparatorChar = '/';
+                newDirectorySeparatorChar = '\\';
+            }
+
+            if (path.Contains(oldDirectorySeparatorChar, StringComparison.Ordinal))
+            {
+                path = path.Replace(oldDirectorySeparatorChar, newDirectorySeparatorChar);
+            }
+
+            if (subPath.Contains(oldDirectorySeparatorChar, StringComparison.Ordinal))
+            {
+                subPath = subPath.Replace(oldDirectorySeparatorChar, newDirectorySeparatorChar);
+            }
+
+            // We have to ensure that the sub path ends with a directory separator otherwise we'll get weird results
+            // when the sub path matches a similar but in-complete subpath
+            if (!subPath.EndsWith(newDirectorySeparatorChar))
+            {
+                subPath += newDirectorySeparatorChar;
+            }
+
+            if (newSubPath.Contains(oldDirectorySeparatorChar, StringComparison.Ordinal))
+            {
+                newSubPath = newSubPath.Replace(oldDirectorySeparatorChar, newDirectorySeparatorChar);
+            }
+
+            if (!newSubPath.EndsWith(newDirectorySeparatorChar))
+            {
+                newSubPath += newDirectorySeparatorChar;
+            }
+
+            if (!path.Contains(subPath, StringComparison.OrdinalIgnoreCase))
+            {
+                newPath = null;
+                return false;
+            }
+
+            newPath = path.Replace(subPath, newSubPath, StringComparison.OrdinalIgnoreCase);
+            return true;
```

Comment:
Would this be more optimised, as I would expect _equals_ to be faster than _contains_ ???
```
            newPath = path.Replace(subPath, newSubPath, StringComparison.OrdinalIgnoreCase);
            return !string.Equals(newPath, path, StringComparison.OrdinalIgnoreCase);
```